control-plane-operator/hostedclusterconfigoperator/controllers/globalps/globalps.go

```diff
@@ -49,37 +51,73 @@ func (r *Reconciler) Reconcile(ctx context.Context, req crreconcile.Request) (cr
 
 // reconcileGlobalPullSecret reconciles the original pull secret given by HCP and merges it with a new pull secret provided by the user.
 // The new pull secret is only stored in the DataPlane side so, it's not exposed in the API. It lives in the kube-system namespace of the DataPlane.
-// If that PS exists, the HCCO deploys a DaemonSet which mounts the whole Root FS of the node, and merges the new PS with the original one.
-// If the PS doesn't exist, the HCCO doesn't do anything.
+// - If that PS is created, the HCCO deploys a DaemonSet which mounts the node's kubeconfig's file, and merges the new PS with the original one.
+// - If the PS doesn't exist, the HCCO doesn't do anything.
+// - If at some point the user deletes the additional pull secret, the daemonSet will not be removed
+// - If the user never added that additional pull secret, the daemonSet will not be deployed in the DataPlane
```

Comment:
From what I read in this globalps.go file, no matter the user adds the additional pull secret or not, the `reconcileDaemonSet` will be called which will create a daemonset, right ?

Reply:
Yeah you're right, let me fix the comment.